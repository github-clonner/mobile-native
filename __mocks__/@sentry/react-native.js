export const captureMessage = jest.fn();
export default {};
