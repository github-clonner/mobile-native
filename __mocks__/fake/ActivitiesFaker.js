
function load(count, container) {

  let activities = [...Array(count)].map((_, i) => {
    const code = 'activityguid' + i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid:code,
      custom_data:false,
      custom_type:false,
      rowKey: 'something' + i,
      description:"Congratulations! ",
      edited:"",
      guid:code,
      mature:false,
      time_created: "1522036284",
      ownerObj:{
        guid: "824853017709780997",
        type: "user",
        subtype: false,
        time_created: "1522036284",
        // getAvatarSource: () => {
        //   return {
        //     source:'http://thisisaurl'
        //   }
        // }
      },
      shouldBeBlured: jest.fn(),
      message:"Message",
      title:'TITLE',
      owner_guid:"824853017709780997",
      parent_guid:"838106762591510528",
      perma_url:false,
      thumbnail_src:false,
      type:"activity",
      wire_totals: {
        tokens: 1000000000000000000
      },
      containerObj: container
      // _list: {
      //   viewed: {
      //     viewed: new Map([["1019155171608096768",true]]),
      //     addViewed: () => {
      //       return;
      //     }
      //   }
      // },
      // getThumbSource: () => {
      //   return {
      //     source:'http://thisisaurl'
      //   }
      // }
    }
  });

  return {
    'activities': activities,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa'
  }

}

/**
 * comment service data faker
 * @param {integer} count
 */
export function activitiesServiceFaker() {
  return { load: load}
}

