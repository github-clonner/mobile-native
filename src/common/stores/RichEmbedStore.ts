//@ts-nocheck
import { observable, action, extendObservable, isObservable } from 'mobx';
import RichEmbedService from '../services/rich-embed.service';
import Util from '../helpers/util';
import logService from '../services/log.service';

/**
 * Rich embed store
 */
export default class RichEmbedStore {
  @observable hasRichEmbed = false;
  @observable metaInProgress = false;
  @observable meta = null;
  richEmbedUrl = '';
  _richEmbedFetchTimer = null;
  setRichEmbedPromise = null; // used for testing

  /**
   * Richembed check
   */
  @action
  richEmbedCheck = (text) => {
    const matches = Util.urlReSingle.exec(text);

    if (this._richEmbedFetchTimer) {
      clearTimeout(this._richEmbedFetchTimer);
    }

    if (matches) {
      const url =
        (!(
          matches[0].startsWith('https://') || matches[0].startsWith('http://')
        )
          ? 'https://'
          : '') + matches[0];

      if (
        !this.hasRichEmbed ||
        (this.hasRichEmbed &&
          url.toLowerCase() !== this.richEmbedUrl.toLowerCase())
      ) {
        this._richEmbedFetchTimer = setTimeout(
          () => (this.setRichEmbedPromise = this.setRichEmbed(url)),
          750,
        );
      }
    }
  };

  /**
   * Clear rich embed
   */
  clearRichEmbedAction = () => {
    this.clearRichEmbed();

    if (this._richEmbedFetchTimer) {
      clearTimeout(this._richEmbedFetchTimer);
    }
  };

  /**
   * Clear rich embed
   */
  @action
  clearRichEmbed = () => {
    this.hasRichEmbed = false;
    this.richEmbedUrl = '';
    this.meta = null;
  };

  /**
   * Set rich embed
   * @param {string} url
   */
  @action
  async setRichEmbed(url) {
    this.hasRichEmbed = true;
    this.richEmbedUrl = url;
    this.meta = null;
    this.metaInProgress = true;

    try {
      const meta = await RichEmbedService.getMeta(url);
      this.meta = meta;
      this.metaInProgress = false;
    } catch (e) {
      this.metaInProgress = false;
      logService.exception('[RichEmbedStore]', e);
    }
  }
}
