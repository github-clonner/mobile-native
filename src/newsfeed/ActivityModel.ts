import { runInAction, action, observable, decorate } from 'mobx';
import FastImage from 'react-native-fast-image';
import { FlatList, Alert } from 'react-native';

import BaseModel from '../common/BaseModel';
import UserModel from '../channel/UserModel';
import wireService from '../wire/WireService';
import sessionService from '../common/services/session.service';
import {
  setPinPost,
  deleteItem,
  unfollow,
  follow,
  update,
} from '../newsfeed/NewsfeedService';
import api from '../common/services/api.service';

import { GOOGLE_PLAY_STORE, MINDS_CDN_URI, MINDS_URI } from '../config/Config';
import i18n from '../common/services/i18n.service';
import logService from '../common/services/log.service';
import entitiesService from '../common/services/entities.service';
import type { ThumbSize } from '../types/Common';
import type GroupModel from 'src/groups/GroupModel';

type Thumbs = Record<ThumbSize, string>;

/**
 * Activity model
 */
export default class ActivityModel extends BaseModel {
  // Observable properties
  @observable pinned: boolean = false;
  @observable time_created: string = '';
  @observable message: string = '';
  @observable title: string = '';
  @observable mature: boolean = false;
  @observable edited: '0' | '1' = '0';

  // decorated observables
  'is:following': boolean;
  'thumbs:down:count': number;
  'thumbs:up:count': number;
  'comments:count': number;
  'thumbs:down:user_guids': Array<number>;
  'thumbs:up:user_guids': Array<number>;

  containerObj?: GroupModel;
  remind_object?: ActivityModel;
  ownerObj!: UserModel;
  listRef?: FlatList<any>;
  thumbnails?: Thumbs;
  paywall: '1' | '' = '';
  paywall_unlocked: boolean = false;
  guid: string = '';
  subtype: string = '';
  entity_guid: string = '';
  owner_guid: string = '';
  custom_type?: string;
  custom_data?: Array<any> | any;
  nsfw?: Array<number>;
  flags?: any;
  reminds: number = 0;
  impressions: number = 0;
  perma_url?: string;
  cinemr_guid?: string;
  thumbnail_src?: string;
  dontPin?: boolean;
  boosted?: boolean;
  wire_threshold?: {
    type: 'token' | 'money';
    min: number;
  } | null;
  _preview?: boolean;
  /**
   * Set programatically to indicate that the parent is marked as mature
   */
  is_parent_mature?: boolean;

  /**
   * Mature visibility flag
   */
  @observable mature_visibility: boolean = false;

  /**
   * Is visible in flat list
   */
  @observable is_visible: boolean = true;

  /**
   *  List reference setter
   */
  set _list(value) {
    this.__list = value;

    // the reminded object need access to the metadata service too
    if (this.remind_object) {
      this.remind_object._list = value;
    }
  }

  /**
   *  List reference getter
   */
  get _list() {
    return this.__list;
  }

  toPlainObject() {
    const plainEntity = super.toPlainObject();
    if (plainEntity.remind_object && plainEntity.remind_object.__list) {
      delete plainEntity.remind_object.__list;
    }

    return plainEntity;
  }

  /**
   * Child models
   */
  childModels() {
    return {
      ownerObj: UserModel,
      remind_object: ActivityModel,
    };
  }

  /**
   * Get the activity thumb source
   * {uri: 'http...'}
   * @param {string} size
   */
  getThumbSource(size: ThumbSize = 'medium') {
    // for gif use always the same size to take adventage of the cache (they are not resized)
    if (this.isGif()) {
      size = 'medium';
    }

    if (this.thumbnails && this.thumbnails[size]) {
      return { uri: this.thumbnails[size], headers: api.buildHeaders() };
    }

    // fallback to old behavior
    if (this.paywall || this.paywall_unlocked) {
      return {
        uri: MINDS_URI + 'fs/v1/thumbnail/' + this.entity_guid,
        headers: api.buildHeaders(),
      };
    }
    if (this.custom_type === 'batch') {
      return {
        uri: MINDS_CDN_URI + 'fs/v1/thumbnail/' + this.entity_guid + '/' + size,
        headers: api.buildHeaders(),
      };
    }
    return {
      uri: MINDS_CDN_URI + 'fs/v1/thumbnail/' + this.guid + '/' + size,
      headers: api.buildHeaders(),
    };
  }

  isGif(): boolean {
    return this.custom_data && this.custom_data[0] && this.custom_data[0].gif;
  }

  /**
   * Preload thumb on image cache
   */
  preloadThumb(size: ThumbSize = 'medium') {
    FastImage.preload([this.getThumbSource(size)]);
  }

  shouldBeBlured(): boolean {
    const user = sessionService.getUser();

    if (user && user.mature) {
      return false;
    }

    if (this.nsfw !== undefined) {
      let res = [1, 2, 4].filter((nsfw) => this.nsfw!.indexOf(nsfw) > -1)
        .length;
      if (res) {
        return true;
      }
    }

    if (typeof this.flags !== 'undefined') {
      return !!this.flags.mature;
    }

    if (typeof this.mature !== 'undefined') {
      return !!this.mature;
    }

    if (
      typeof this.custom_data !== 'undefined' &&
      typeof this.custom_data[0] !== 'undefined'
    ) {
      return !!this.custom_data[0].mature;
    }

    if (this.custom_data !== undefined) {
      return !!this.custom_data.mature;
    }

    return false;
  }

  /**
   * Get activity text
   */
  get text() {
    return this.message || '';
  }

  @action
  toggleMatureVisibility() {
    if (GOOGLE_PLAY_STORE) {
      return;
    }
    this.mature_visibility = !this.mature_visibility;

    if (this.remind_object && this.remind_object.mature) {
      this.remind_object.mature_visibility = this.mature_visibility;
    }
  }

  @action
  setVisible(value: boolean) {
    this.is_visible = value;
  }

  /**
   * Increment the comments counter
   */
  @action
  incrementCommentsCounter() {
    this['comments:count']++;
  }

  /**
   * Decrement the comments counter
   */
  @action
  decrementCommentsCounter() {
    this['comments:count']--;
  }

  /**
   * Unlock the activity and update data on success
   */
  @action
  async unlock(ignoreError = false) {
    try {
      const result: object | false = await wireService.unlock(this.guid);
      if (result) {
        // all changes should be atomic (trigger render only once)
        runInAction(() => {
          // create a new model because we need the child models
          const model: ActivityModel = ActivityModel.create(result);
          Object.assign(this, model);
        });
      }
      return result;
    } catch (err) {
      if (!ignoreError) {
        Alert.alert(err.message);
      }
      return false;
    }
  }

  @action
  async togglePin() {
    // allow owners only
    if (!this.isOwner()) {
      return;
    }

    try {
      this.pinned = !this.pinned;
      await setPinPost(this.guid, this.pinned);
    } catch (e) {
      this.pinned = !this.pinned;
      Alert.alert(i18n.t('errorPinnedPost'));
    }
  }

  @action
  async deleteEntity() {
    try {
      await deleteItem(this.guid);
      this.removeFromList();
      entitiesService.deleteFromCache(this.urn);
    } catch (err) {
      logService.exception('[ActivityModel]', err);
      throw err;
    }
  }

  @action
  async toggleFollow() {
    const method = this['is:following'] ? unfollow : follow;
    try {
      await method(this.guid);
      runInAction(() => {
        this['is:following'] = !this['is:following'];
      });
    } catch (err) {
      logService.exception('[OffsetFeedListStore]', err);
      throw err;
    }
  }

  @action
  async updateActivity(data: any = {}) {
    const entity: any = this.toPlainObject();

    if (data) {
      for (const field in data) {
        entity[field] = data[field];
      }
    }

    // call update endpoint
    await update(entity);

    // update instance properties
    this.update(data);

    this.setEdited(entity.message);
  }

  @action
  setEdited(message: string) {
    this.message = message;
    this.edited = '1';
  }
}

/**
 * Define model observables
 */
decorate(ActivityModel, {
  //@ts-ignore
  'thumbs:down:count': observable,
  'thumbs:up:count': observable,
  'comments:count': observable,
  'thumbs:down:user_guids': observable,
  'thumbs:up:user_guids': observable,
});
