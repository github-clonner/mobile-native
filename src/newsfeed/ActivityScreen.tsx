import React, { Component } from 'react';

import { View, Text } from 'react-native';
import { observer } from 'mobx-react';
import FastImage from 'react-native-fast-image';

import { CommonStyle as CS } from '../styles/Common';
import CommentList from '../comments/CommentList';
import Activity from '../newsfeed/activity/Activity';
import ActivityModel from '../newsfeed/ActivityModel';
import { ComponentsStyle } from '../styles/Components';
import SingleEntityStore from '../common/stores/SingleEntityStore';
import CenteredLoading from '../common/components/CenteredLoading';
import commentsStoreProvider from '../comments/CommentsStoreProvider';
import i18n from '../common/services/i18n.service';
import OffsetFeedListStore from '../common/stores/OffsetFeedListStore';
import { FLAG_VIEW } from '../common/Permissions';
import type CommentsStore from 'src/comments/CommentsStore';

type PropsType = {
  route: any;
  navigation: any;
};

/**
 * Activity screen
 */
@observer
class ActivityScreen extends Component<PropsType> {
  comments: CommentsStore;
  activity: Activity | null = null;

  /**
   * Store
   */
  entityStore: SingleEntityStore<ActivityModel> = new SingleEntityStore();

  /**
   * Constructor
   * @param {object} props
   */
  constructor(props) {
    super(props);

    this.comments = commentsStoreProvider.get();

    this.loadEntity();
  }

  /**
   * Load entity
   */
  async loadEntity() {
    const params = this.props.route.params;

    if (params.entity && (params.entity.guid || params.entity.entity_guid)) {
      const urn =
        'urn:entity:' + (params.entity.guid || params.entity.entity_guid);

      const entity = ActivityModel.checkOrCreate(params.entity);

      if (!entity.can(FLAG_VIEW, true)) {
        this.props.navigation.goBack();
        return;
      }

      this.entityStore.loadEntity(urn, entity, true);

      // change metadata source
      if (params.entity._list && params.entity._list.metadataService) {
        params.entity._list.metadataService.pushSource('single');
      }
    } else {
      const urn = 'urn:entity:' + params.guid;
      await this.entityStore.loadEntity(urn);

      if (!this.entityStore.entity?.can(FLAG_VIEW, true)) {
        this.props.navigation.goBack();
        return;
      }
    }

    if (params.entity && params.entity._list) {
      // this second condition it's for legacy boost feed
      if (params.entity._list instanceof OffsetFeedListStore) {
        params.entity._list.addViewed(params.entity);
      } else {
        params.entity._list.viewed.addViewed(
          params.entity,
          params.entity._list.metadataService,
        );
      }
    }
  }

  /**
   * Component will unmount
   */
  componentWillUnmount() {
    const entity = this.entityStore.entity;

    if (entity && entity._list && entity._list.metadataService) {
      entity._list.metadataService.popSource();
    }
  }

  /**
   * Get header
   */
  getHeader() {
    return this.entityStore.entity ? (
      <Activity
        ref={this.setRef}
        entity={this.entityStore.entity}
        navigation={this.props.navigation}
        autoHeight={false}
      />
    ) : null;
  }

  /**
   * Set ref
   * @param {NodeRef}
   */
  setRef = (o) => (this.activity = o);

  /**
   * On comment input focus
   */
  onFocus = () => {
    this.activity?.pauseVideo();
  };

  /**
   * Render
   */
  render() {
    if (!this.entityStore.entity && !this.entityStore.errorLoading) {
      return <CenteredLoading />;
    }

    if (!this.entityStore.entity?.can(FLAG_VIEW, true)) {
      this.props.navigation.goBack();
      return null;
    }

    return (
      <View style={[CS.flexContainer]}>
        {!this.entityStore.errorLoading ? (
          //@ts-ignore user store is injected
          <CommentList
            header={this.getHeader()}
            entity={this.entityStore.entity}
            store={this.comments}
            navigation={this.props.navigation}
            onInputFocus={this.onFocus}
            route={this.props.route}
          />
        ) : (
          <View style={CS.flexColumnCentered}>
            <FastImage
              resizeMode={FastImage.resizeMode.contain}
              style={ComponentsStyle.logo}
              source={require('../assets/logos/logo.png')}
            />
            <Text style={[CS.fontL, CS.colorDanger]}>
              {i18n.t('activity.error')}
            </Text>
            <Text style={[CS.fontM]}>{i18n.t('activity.tryAgain')}</Text>
          </View>
        )}
      </View>
    );
  }
}

export default ActivityScreen;
