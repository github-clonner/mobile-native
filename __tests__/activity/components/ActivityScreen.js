import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

import ActivityScreen from '../../../src/newsfeed/ActivityScreen';
import RichEmbedStore from '../../../src/common/stores/RichEmbedStore';
import commentsStoreProvider from '../../../src/comments/CommentsStoreProvider';

import { commentsServiceFaker } from '../../../__mocks__/fake/CommentsFaker';
import { activitiesServiceFaker } from '../../../__mocks__/fake/ActivitiesFaker';
import CommentList from '../../../src/comments/CommentList';
import entitiesService from '../../../src/common/services/entities.service';
import ActivityModel from '../../../src/newsfeed/ActivityModel';

jest.mock('../../../src/newsfeed/NewsfeedService');
jest.mock('../../../src/newsfeed/activity/Activity', () => 'Activity');
jest.mock('../../../src/comments/CommentList', () => 'CommentList');
jest.mock(
  '../../../src/common/components/CenteredLoading',
  () => 'CenteredLoading',
);
jest.mock('../../../src/comments/CommentsStore');
jest.mock('../../../src/comments/CommentsStoreProvider');
jest.mock('../../../src/common/services/entities.service');

describe('Activity screen component', () => {
  let screen, navigation, route;
  beforeEach(() => {
    let mockResponse = commentsServiceFaker().load(5);
    commentsStoreProvider.get.mockReturnValue({
      comments: mockResponse.comments,
      embed: new RichEmbedStore(),
      loadNext: mockResponse['load-next'],
      loadPrevious: mockResponse['load-previous'],
      setText: () => {},
      post: () => {},
      loadComments: () => {
        return mockResponse.comments;
      },
      attachment: {},
    });
  });

  it('renders correctly with an entity as param', async () => {
    navigation = {
      push: jest.fn(),
    };

    route = {
      routeName: 'some',
      params: { entity: activitiesServiceFaker().load(1).activities[0] },
    };

    entitiesService.single.mockResolvedValue(
      ActivityModel.create(route.params.entity),
    );

    screen = shallow(<ActivityScreen navigation={navigation} route={route} />);

    // shoul show loading
    expect(screen).toMatchSnapshot();

    // unmount
    await screen.instance().loadEntity();

    jest.runAllTicks();

    // await is important here!
    await screen.update();

    // should show the activity
    expect(screen).toMatchSnapshot();
    // should have a comment list component
    expect(screen.find(CommentList)).toHaveLength(1);
  });
});
