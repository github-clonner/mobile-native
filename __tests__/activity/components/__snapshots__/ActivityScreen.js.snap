// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity screen component renders correctly with an entity as param 1`] = `<CenteredLoading />`;

exports[`Activity screen component renders correctly with an entity as param 2`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
    ]
  }
>
  <CommentList
    entity={
      ActivityModel {
        "__list": null,
        "allow_comments": true,
        "attachment_guid": false,
        "blurb": false,
        "comments:count": undefined,
        "containerObj": undefined,
        "container_guid": "activityguid0",
        "custom_data": false,
        "custom_type": false,
        "description": "Congratulations! ",
        "edited": "",
        "entity_guid": "",
        "guid": "activityguid0",
        "impressions": 0,
        "isOwner": [Function],
        "is_visible": true,
        "mature": false,
        "mature_visibility": false,
        "message": "Message",
        "ownerObj": UserModel {
          "__list": null,
          "admin": false,
          "confirmEmail": [Function],
          "eth_wallet": "",
          "founder": false,
          "guid": "824853017709780997",
          "isOwner": [Function],
          "last_accepted_tos": 0,
          "mature": false,
          "plus": false,
          "rewards": false,
          "subtype": false,
          "time_created": "1522036284",
          "type": "user",
          "urn": "",
          "username": "",
          "verified": false,
        },
        "owner_guid": "824853017709780997",
        "parent_guid": "838106762591510528",
        "paywall": "",
        "paywall_unlocked": false,
        "perma_url": false,
        "permissions": Object {},
        "pinned": false,
        "reminds": 0,
        "rowKey": "something0",
        "shouldBeBlured": [MockFunction],
        "subtype": "",
        "thumbnail_src": false,
        "thumbs:down:count": undefined,
        "thumbs:down:user_guids": undefined,
        "thumbs:up:count": undefined,
        "thumbs:up:user_guids": undefined,
        "time_created": "1522036284",
        "title": "TITLE",
        "type": "activity",
        "urn": "",
        "username": "",
        "wire_totals": Object {
          "tokens": 1000000000000000000,
        },
      }
    }
    header={
      <Activity
        autoHeight={false}
        entity={
          ActivityModel {
            "__list": null,
            "allow_comments": true,
            "attachment_guid": false,
            "blurb": false,
            "comments:count": undefined,
            "containerObj": undefined,
            "container_guid": "activityguid0",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "entity_guid": "",
            "guid": "activityguid0",
            "impressions": 0,
            "isOwner": [Function],
            "is_visible": true,
            "mature": false,
            "mature_visibility": false,
            "message": "Message",
            "ownerObj": UserModel {
              "__list": null,
              "admin": false,
              "confirmEmail": [Function],
              "eth_wallet": "",
              "founder": false,
              "guid": "824853017709780997",
              "isOwner": [Function],
              "last_accepted_tos": 0,
              "mature": false,
              "plus": false,
              "rewards": false,
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
              "urn": "",
              "username": "",
              "verified": false,
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "paywall": "",
            "paywall_unlocked": false,
            "perma_url": false,
            "permissions": Object {},
            "pinned": false,
            "reminds": 0,
            "rowKey": "something0",
            "shouldBeBlured": [MockFunction],
            "subtype": "",
            "thumbnail_src": false,
            "thumbs:down:count": undefined,
            "thumbs:down:user_guids": undefined,
            "thumbs:up:count": undefined,
            "thumbs:up:user_guids": undefined,
            "time_created": "1522036284",
            "title": "TITLE",
            "type": "activity",
            "urn": "",
            "username": "",
            "wire_totals": Object {
              "tokens": 1000000000000000000,
            },
          }
        }
        navigation={
          Object {
            "push": [MockFunction],
          }
        }
      />
    }
    navigation={
      Object {
        "push": [MockFunction],
      }
    }
    onInputFocus={[Function]}
    route={
      Object {
        "params": Object {
          "entity": Object {
            "attachment_guid": false,
            "blurb": false,
            "containerObj": undefined,
            "container_guid": "activityguid0",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "guid": "activityguid0",
            "mature": false,
            "message": "Message",
            "ownerObj": Object {
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "perma_url": false,
            "rowKey": "something0",
            "shouldBeBlured": [MockFunction],
            "thumbnail_src": false,
            "time_created": "1522036284",
            "title": "TITLE",
            "type": "activity",
            "wire_totals": Object {
              "tokens": 1000000000000000000,
            },
          },
        },
        "routeName": "some",
      }
    }
    store={
      Object {
        "attachment": Object {},
        "comments": Array [
          Object {
            "attachment_guid": false,
            "blurb": false,
            "container_guid": "guid0",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "guid": "guid0",
            "mature": false,
            "ownerObj": Object {
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "perma_url": false,
            "thumbnail_src": false,
            "type": "comment",
          },
          Object {
            "attachment_guid": false,
            "blurb": false,
            "container_guid": "guid1",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "guid": "guid1",
            "mature": false,
            "ownerObj": Object {
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "perma_url": false,
            "thumbnail_src": false,
            "type": "comment",
          },
          Object {
            "attachment_guid": false,
            "blurb": false,
            "container_guid": "guid2",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "guid": "guid2",
            "mature": false,
            "ownerObj": Object {
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "perma_url": false,
            "thumbnail_src": false,
            "type": "comment",
          },
          Object {
            "attachment_guid": false,
            "blurb": false,
            "container_guid": "guid3",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "guid": "guid3",
            "mature": false,
            "ownerObj": Object {
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "perma_url": false,
            "thumbnail_src": false,
            "type": "comment",
          },
          Object {
            "attachment_guid": false,
            "blurb": false,
            "container_guid": "guid4",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "guid": "guid4",
            "mature": false,
            "ownerObj": Object {
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "perma_url": false,
            "thumbnail_src": false,
            "type": "comment",
          },
        ],
        "embed": RichEmbedStore {
          "_richEmbedFetchTimer": null,
          "clearRichEmbedAction": [Function],
          "richEmbedUrl": "",
          "setRichEmbedPromise": null,
        },
        "loadComments": [Function],
        "loadNext": "aaaaaa",
        "loadPrevious": "aaaaaa",
        "post": [Function],
        "setText": [Function],
      }
    }
  />
</View>
`;
